support-workers/src/main/scala/com/gu/support/workers/lambdas/UpdateSupporterProductData.scala

```diff
@@ -65,61 +76,88 @@ object UpdateSupporterProductData {
   def getSupporterRatePlanItemFromState(
       state: SendThankYouEmailState,
       catalogService: CatalogService,
-  ): Either[Unit, Option[SupporterRatePlanItem]] =
+  ): Either[String, Option[SupporterRatePlanItem]] =
     state match {
       case SendThankYouEmailContributionState(user, product, _, _, subscriptionNumber) =>
         catalogService
           .getProductRatePlan(Contribution, product.billingPeriod, NoFulfilmentOptions, NoProductOptions)
           .map(productRatePlan =>
-            supporterRatePlanItem(
-              subscriptionName = subscriptionNumber,
-              identityId = user.id,
-              productRatePlanId = productRatePlan.id,
-              productRatePlanName = s"support-workers added ${product.describe}",
-              Some(ContributionAmount(product.amount, product.currency.iso)),
+            Some(
+              supporterRatePlanItem(
+                subscriptionName = subscriptionNumber,
+                identityId = user.id,
+                productRatePlanId = productRatePlan.id,
+                productRatePlanName = s"support-workers added ${product.describe}",
+                Some(ContributionAmount(product.amount, product.currency.iso)),
+              ),
             ),
           )
-          .toRight(())
-          .map(Some(_))
+          .toRight(s"Unable to create SupporterRatePlanItem from state $state")
```

Comment:
I found this quite hard to follow in github because it creates nested Options then turns the outer one into an Either! Perhaps swapping them round might help
```
.getProductRatePlan(...
.toRight(...
.map(...
```

Reply:
In the way it was previously? I changed it to this because I thought it was clearer!

Reply:
yeah, we must reason about this in different ways!